### PR TITLE
Task/WG-10: Add projects and user queries.

### PR DESCRIPTION
**Pull request by duckonomy**
## Overview: ##
Adds basic project and user queries for react.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WG-10](https://jira.tacc.utexas.edu/browse/WG-10)

## Summary of Changes: ##
Added basic query to userinfo and projects to production geoapi.
Also added slice for projects and modified `authSlice` to include user information.
Added prettier in npm script.

## Testing Steps: ##
1. NOTE: Testing is set with production geoapi.
2. Open page (http://localhost:4200)
3. Ensure in chrome networks, you see the projects and userinfo responses.

## Notes: ##
I've tried setting locally with the jwt.. For some reason I get a CORS error for endpoints that should have that covered.

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
         "test": "jest",
         "lint": "npm run lint:js",
         "lint:js": "eslint . --ext .js,.jsx,.ts,.tsx",
-        "lint:fix": "eslint . --ext .js,.jsx,.ts,.tsx --fix"
+        "lint:fix": "eslint . --ext .js,.jsx,.ts,.tsx --fix",
+        "prettier:check": "prettier --single-quote --check src",
```
Comment (nathanfranklin) [inline react/package.json:14]: does this need to be added to CI?

File: react/src/redux/api/geoapi.ts [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
-        headers.set('Authorization', `Bearer ${token}`);
+        headers.set('Authorization', `Bearer ${token.token}`);
       }
+
       headers.set('Content-Type', 'application/json;charset=UTF-8');
-      headers.set('Authorization', 'anonymous');
+
+      // TODO below adding of JWT if localhost and then add JWT
+      // we put the JWT on the request to our geoapi API because it is not behind ws02 if in local dev
+      // and if user is logged in
```
Comment (nathanfranklin) [inline react/src/redux/api/geoapi.ts:23]:
```suggestion
```
duplicated comment

File: react/src/redux/api/geoapi.ts [hunk 3]
```diff
@@ -1,21 +1,23 @@
 import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/dist/query/react';
-import store from '../store';
+import type { RootState } from '../store';
 
 // TODO: make configurable so can be https://agave.designsafe-ci.org/geo-staging/v2 or https://agave.designsafe-ci.org/geo/v2
-const BASE_URL = 'https:localhost:8888';
+const BASE_URL = 'https://agave.designsafe-ci.org/geo/v2';
```
Comment (nathanfranklin) [inline react/src/redux/api/geoapi.ts:5]:
```suggestion
// See https://tacc-main.atlassian.net/browse/WG-196
const BASE_URL = 'https://agave.designsafe-ci.org/geo/v2';
```

File: react/package.json (hunk 1 above)
Comment (nathanfranklin) [reply react/package.json:14]: adding a ticket: https://tacc-main.atlassian.net/browse/WG-197

**Review by nathanfranklin — APPROVED** [review]
LGTM 👍

**Review by sophia-massie — APPROVED** [review]
LGTM!

**Pull request merged**